tests/test_views.py

```diff
@@ -0,0 +1,223 @@
+        }
+        self.assertEqual(got, want)
+
+    def test_pending(self):
+        """Pending renewal are renewable, if current and actiomable."""
+        advantage = make_advantage()
+        now = datetime.now(timezone.utc)
+        start = (now - timedelta(days=1)).isoformat()
+        end = (now + timedelta(days=1)).isoformat()
+        contract_info = {
+            "renewals": [
+                {
+                    "id": "1",
+                    "status": "pending",
+                    "actionable": True,
+                    "start": start,
+                    "end": end,
+                }
+            ],
+        }
+        got = views.make_renewal(advantage, contract_info)
+        want = {
+            "id": "1",
+            "status": "pending",
+            "actionable": True,
+            "start": start,
+            "end": end,
+            "renewable": True,
+        }
+        self.assertEqual(got, want)
+
+    def test_closed(self):
+        """Closed renewals are not renewable."""
+        advantage = make_advantage()
+        now = datetime.now(timezone.utc)
+        start = (now - timedelta(days=1)).isoformat()
+        end = (now + timedelta(days=1)).isoformat()
+        contract_info = {
+            "renewals": [
+                {
+                    "id": "1",
+                    "status": "closed",
+                    "actionable": True,
+                    "start": start,
+                    "end": end,
+                }
+            ],
+        }
+        got = views.make_renewal(advantage, contract_info)
+        want = {
+            "id": "1",
+            "status": "closed",
+            "actionable": True,
+            "start": start,
+            "end": end,
+            "renewable": False,
+        }
+        self.assertEqual(got, want)
+
+    def test_processing_renewable(self):
+        """Processing renewals are renewable if they require payment method."""
+        now = datetime.now(timezone.utc)
+        start = (now - timedelta(days=1)).isoformat()
+        end = (now + timedelta(days=1)).isoformat()
+        renewal = {
+            "id": "1",
+            "status": "processing",
+            "actionable": True,
+            "start": start,
+            "end": end,
+            "stripeInvoices": [
+                {
+                    "pi_status": "requires_payment_method",
+                    "subscription_status": "incomplete_expired",
+                }
+            ],
+        }
+        advantage = make_advantage(renewal=renewal)
+        contract_info = {"renewals": [renewal]}
+        got = views.make_renewal(advantage, contract_info)
+        want = {
+            "id": "1",
+            "status": "processing",
+            "actionable": True,
+            "start": start,
+            "end": end,
+            "stripeInvoices": [
+                {
+                    "pi_status": "requires_payment_method",
+                    "subscription_status": "incomplete_expired",
```

Comment:
A follow on from my previous comment - attempting to renew when the invoice has `incomplete_expired` status has always failed for me when testing against the contracts service:

```suggestion
                    "subscription_status": "incomplete",
```

Reply:
Done.